test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/BulkOperation/BulkInsertController.cs

```diff
@@ -0,0 +1,300 @@
+                        LikeMost=Sport.Pingpong|Sport.Basketball,
+                        Like=new List<Sport>{Sport.Pingpong,Sport.Basketball}
+                    },
+                    Friends = new List<Friend>{new Friend { Id=1,Name="Test0"} ,new Friend { Id=2,Name="Test1"} }
+                },
+            };
+        }
+
+        [EnableQuery(PageSize = 10, MaxExpansionDepth = 5)]
+        public ITestActionResult Get()
+        {
+            return Ok(Employees.AsQueryable());
+        }
+
+        public ITestActionResult Get(int key)
+        {
+            var emp = Employees.SingleOrDefault(e => e.ID == key);
+            return Ok(Employees.SingleOrDefault(e => e.ID == key));
+        }
+
+        [ODataRoute("Employees({key})/Friends")]
+        public ITestActionResult GetFriends(int key)
+        {
+            var emp = Employees.SingleOrDefault(e => e.ID == key);
+            return Ok(emp.Friends);
+        }
+
+        public ITestActionResult GetAccessLevelFromEmployee(int key)
+        {
+            return Ok(Employees.SingleOrDefault(e => e.ID == key).AccessLevel);
+        }
+
+        public ITestActionResult GetNameFromEmployee(int key)
+        {
+            return Ok(Employees.SingleOrDefault(e => e.ID == key).Name);
+        }
+
+        [EnableQuery]
+        public ITestActionResult GetSkillSetFromEmployee(int key)
+        {
+            return Ok(Employees.SingleOrDefault(e => e.ID == key).SkillSet);
+        }
+
+        [EnableQuery]
+        public ITestActionResult GetFavoriteSportsFromEmployee(int key)
+        {
+            var employee = Employees.SingleOrDefault(e => e.ID == key);
+            return Ok(employee.FavoriteSports);
+        }
+
+        [HttpGet]
+        [ODataRoute("Employees({key})/FavoriteSports/LikeMost")]
+        public ITestActionResult GetFavoriteSportLikeMost(int key)
+        {
+            var firstOrDefault = Employees.FirstOrDefault(e => e.ID == key);
+            return Ok(firstOrDefault.FavoriteSports.LikeMost);
+        }
+
+        public ITestActionResult Post([FromBody]Employee employee)
+        {
+            employee.ID = Employees.Count + 1;
+            Employees.Add(employee);
+
+            return Created(employee);
+        }
+
+        [ODataRoute("Employees({key})/FavoriteSports/LikeMost")]
+        public ITestActionResult PostToSkillSet(int key, [FromBody]Skill newSkill)
+        {
+            Employee employee = Employees.FirstOrDefault(e => e.ID == key);
+            if (employee == null)
+            {
+                return NotFound();
+            }
+            employee.SkillSet.Add(newSkill);
+            return Updated(employee.SkillSet);
+        }
+
+        [ODataRoute("Employees")]
+        [HttpPatch]
+        public ITestActionResult PatchEmployees( [FromBody] EdmChangedObjectCollection<Employee> coll)
+        {
+            Assert.NotNull(coll);
+            coll.Patch(Employees);
+
+            return Ok(Employees);
```

Comment:
>return Ok(Employees) [](start = 12, length = 20)

This isn't right -- you just want to return the members of the EdmChangedObjectCollection<Employee> as a ChangedObjectCollection, showing what succeeded and what failed, not return all of the employees.